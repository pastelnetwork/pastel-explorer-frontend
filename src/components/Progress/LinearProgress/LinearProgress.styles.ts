import styled from 'styled-components';
import {
  LinearProgress as MuiLinearProgress,
  Typography as MuiTypography,
} from '@mui/material';

export const LinearProgress = styled(MuiLinearProgress)`
  height: 14px;
  width: 120px;
  border-radius: 3px;
`;

export const Typography = styled(MuiTypography)`
  margin-left: 3px;
  font-size: 0.9rem;
  width: 20px;
`;
