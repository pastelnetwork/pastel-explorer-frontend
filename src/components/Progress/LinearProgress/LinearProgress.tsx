import * as React from 'react';
import { Grid } from '@mui/material';

import * as Styles from './LinearProgress.styles';

interface LinearProgressProps {
  value: number;
  description?: string;
}

const LinearProgressComponent: React.FC<LinearProgressProps> = ({ value, description }) => {
  return (
    <Grid container sx={{ alignItems: 'center', justifyContent: 'center' }}>
      <Styles.LinearProgress value={value} variant="determinate" color="primary" />
      {description && <Styles.Typography>{description}</Styles.Typography>}
    </Grid>
  );
};

LinearProgressComponent.defaultProps = {
  description: '',
}

export default LinearProgressComponent;
