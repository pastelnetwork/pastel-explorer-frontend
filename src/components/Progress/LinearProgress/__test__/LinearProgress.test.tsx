import { shallow } from 'enzyme';

import 'jest-styled-components';

import LinearProgress from '../LinearProgress';

describe('components/Progress/LinearProgress', () => {
  const wrapper = shallow(<LinearProgress value={50} />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
