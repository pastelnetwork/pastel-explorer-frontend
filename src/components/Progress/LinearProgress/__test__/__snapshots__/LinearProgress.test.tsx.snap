// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Progress/LinearProgress renders correctly 1`] = `ShallowWrapper {}`;
